src/pages/transactions/edit/Edit.tsx

```diff
@@ -0,0 +1,231 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { FormEvent, useEffect } from 'react';
+import { Card, Element } from '@invoiceninja/cards';
+import { InputField, SelectField } from '@invoiceninja/forms';
+import { transactionTypes } from 'common/constants/transactions';
+import { useCurrencies } from 'common/hooks/useCurrencies';
+import { useTitle } from 'common/hooks/useTitle';
+import { endpoint } from 'common/helpers';
+import {
+  TransactionInput,
+  TransactionResponse,
+} from 'common/interfaces/transactions';
+import { Container } from 'components/Container';
+import { DebouncedCombobox } from 'components/forms/DebouncedCombobox';
+import { Default } from 'components/layouts/Default';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TransactionValidation } from '../common/validation/ValidationInterface';
+import { request } from 'common/helpers/request';
+import { useNavigate, useParams } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { AxiosError } from 'axios';
+import { useTransactionQuery } from '../common/queries';
+import { DecimalNumberInput } from 'components/forms/DecimalNumberInput';
+import { DecimalInputSeparators } from 'common/interfaces/decimal-number-input-separators';
+import { useResolveCurrency } from 'common/hooks/useResolveCurrency';
+
+export function Edit() {
+  const { t } = useTranslation();
+
+  const navigate = useNavigate();
+
+  const { id } = useParams<string>();
+
+  const currencies = useCurrencies();
+
+  const resolveCurrency = useResolveCurrency();
+
+  const { documentTitle } = useTitle('edit_transaction');
+
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const [errors, setErrors] = useState<TransactionValidation>();
+
+  const { data: response } = useTransactionQuery({ id });
+
+  const [transaction, setTransaction] = useState<TransactionInput>();
+
+  const [currencySeparators, setCurrencySeparators] = useState<
+    DecimalInputSeparators | undefined
+  >();
+
+  const pages = [
+    { name: t('transactions'), href: '/transactions' },
+    { name: t('edit_transaction'), href: `/transactions/${id}/edit` },
```

Comment:
Make sure to use `route` helper when dealing with replaced params.

Reply:
> Make sure to use `route` helper when dealing with replaced params.

changed